# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/wasm/ref-types/externref-global-prebarrier.js
# Commit: 89db2f360b8f
# Full Hash: 89db2f360b8fd68c398c70a8c556b92b2058ab00
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2022-03-24 21:54:20
# Regressor Bug: 1642412
# File Overlap Count: 7
# Description:
#   Bug 1642412 - wasm: Implement a 'precise' post-write barrier in baseline. r=jseward
#   
#   Add a post-write barrier which will remove a previous store buffer entry if the
#   new value stored in the field would not require it. This requires the previous
#   value to be loaded before the store and propagated to the post-write barrier
# ==============================================================================

diff -r eabb888a0851 -r 89db2f360b8f js/src/jit-test/tests/wasm/ref-types/externref-global-prebarrier.js
--- a/js/src/jit-test/tests/wasm/ref-types/externref-global-prebarrier.js	Thu Mar 24 17:56:52 2022 +0000
+++ b/js/src/jit-test/tests/wasm/ref-types/externref-global-prebarrier.js	Thu Mar 24 17:56:52 2022 +0000
@@ -34,11 +34,7 @@
 assertEqPreciseStacks(
     endProfiling(),
     [
-        // Expected output for (simulator+via-Ion).
-        ['', '!>', '0,!>', '<,0,!>', 'filtering GC postbarrier,0,!>',
-         '<,0,!>', '0,!>', '!>', ''],
-
-        // Expected output for (simulator+baseline).
+        // Expected output for (simulator+via-Ion/baseline).
         ['', '!>', '0,!>', '<,0,!>', 'GC postbarrier,0,!>',
          '<,0,!>', '0,!>', '!>', ''],
 
@@ -55,8 +51,8 @@
 assertEqPreciseStacks(
     endProfiling(),
     [
-        // Expected output for (simulator+via-Ion).
-        ['', '!>', '0,!>', '', '0,!>', '<,0,!>', 'filtering GC postbarrier,0,!>',
+        // Expected output for (simulator+via-Ion/baseline).
+        ['', '!>', '0,!>', '', '0,!>', '<,0,!>', 'GC postbarrier,0,!>',
          '<,0,!>', '0,!>', '!>', ''],
 
         // Expected output for other configurations.